src/components/Header/NavDropdown/index.tsx

```diff
@@ -30,17 +30,26 @@ const NavDropdownItem: React.VFC<NavDropdownItemType> = ({
   href,
   text,
   to,
-}) => (
-  <MenuItem
-    as={to && Link}
-    to={to}
-    href={href}
-    data-cy={itemDataCy}
-    onClick={closeMenu}
-  >
-    {text}
-  </MenuItem>
-);
+}) => {
+  const isLink = to !== undefined;
+  const linkProps = isLink
+    ? {
+        to,
+      }
+    : {
+        href,
+      };
```

Comment:
This was necessary so that the fields had exclusive types. If we tried to inline this typescript would complain because it wouldn't have a guarantee that to and href aren't used together.